src/readfish/_utils.py

```diff
@@ -449,6 +453,48 @@ def get_device(
             return position
     raise ValueError(f"Could not find device {device!r}")
 
+def get_version(
+        host: str = "127.0.0.1", port: int = None, logger: logging.Logger = None
+) -> str:
+    """Get the version of MinKNOW
+
+    :param host: The host the RPC is listening on, defaults to "127.0.0.1"
+    :param port: The port the RPC is listening on, defaults to None
+    :raises ValueError: If the version of Readfish is incompatible with the version of MinKNOW.
+    :raises Warning: If the version of Readfish may need to be updated soon.
+    :return: The version of MinKNOW
+    """
+    manager = Manager(host=host, port=port)
+    minknow_version = parse_version(manager.core_version)
+    # Check compatability
+    for cv, direction in compatible_version:
+        ver_cv = parse_version(cv)
+        if direction == "gte":
+            if minknow_version <= ver_cv:
+                raise ValueError(
+                    f"Compatability check failed. Readfish version {readfish_version} is not compatible with MinKNOW version {minknow_version}. You need to update readfish."
+                )
+        elif direction == "lte":
+            if minknow_version >= ver_cv:
+                raise ValueError(
+                    f"Compatability check failed. Readfish version {readfish_version} is not compatible with MinKNOW version {minknow_version}. You need to downgrade readfish."
+                )
```

Comment:
This can be deduplicated to:
```python
if (direction == "gte" and minknow_version <= ver_cv) or (direction == "lte" and minknow_version >= ver_cv):
    action = "upgrade" if direction == "gte" else "downgrade"
    message =  f"Compatibility check failed. Readfish version {readfish_version} is not compatible with MinKNOW version {minknow_version}. You need to {action} readfish."
    raise ValueError(
          message
    )
```